tests/suites/test_suite_x509write.function

```diff
@@ -74,6 +75,26 @@ cleanup:
 }
 #endif /* MBEDTLS_USE_PSA_CRYPTO && MBEDTLS_PEM_WRITE_C && MBEDTLS_X509_CSR_WRITE_C */
 
+#if defined(MBEDTLS_X509_CSR_WRITE_C)
+static int csr_set_extended_key_usage( mbedtls_x509write_csr *ctx,
+                        const char *oid, size_t oid_len )
+{
+    unsigned char buf[30] = {0};
```

Comment:
Can we use suitable macro or add comment explaining why 30 bytes is used?

Reply:
You're right, sorry for this.
TBH I set it empirically during the development, but now I added:

- a comment explaining the size of this buffer
- set the size accordingly (I also verified manually that this is the correct value by trying different builds)

Reply:
@mpg @mprse 
This change also impacts PR #6509. Once we're OK with this one, I'll apply the same changes to than one